qbx_vehiclepush/client.lua

```diff
@@ -1,83 +1,208 @@
+
+                if HasEntityCollidedWithAnything(vehicle) then
+                    SetVehicleOnGroundProperly(vehicle)
+                end
+            end
+
+            if DoesEntityExist(vehicle) and NetworkGetEntityOwner(vehicle) ~= playerId and pushingControl then -- handle changing owner in the middle of pushing
+                TriggerServerEvent('qbx_vehiclepush:server:push', {
+                    direction = oldDirection,
+                    netId = VehToNet(vehicle)
+                })
+            end
+        end)
+    end
+
+    pushingControl = value
+end
+
+---@param vehicle number
+local function isVehicleValid(vehicle)
+    local engineHealth = GetVehicleEngineHealth(vehicle)
+
+    return ((engineHealth >= 0 and engineHealth <= config.damageNeeded) or (Entity(vehicle).state.fuel or 100) < 3) and IsVehicleSeatFree(vehicle, -1) and checkClass(vehicle)
+end
+
+---@param vehicle number
+local function vehicleValidityThread(vehicle)
+    CreateThread(function()
+        while pressed.e and pressed.shift do
+            Wait(500)
+            if not isVehicleValid(vehicle) then
+                pressed.e = false
+                pressed.shift = false
+            end
+        end
+    end)
+end
+
+---comment
+---@param vehicle number
+---@param direction direction
+local function taskControlVehicle(vehicle, direction)
+    if NetworkGetEntityOwner(vehicle) == cache.playerId then
+        vehicleControl(vehicle, direction)
+    else
+        TriggerServerEvent('qbx_vehiclepush:server:push', {
+            direction = direction,
+            netId = VehToNet(vehicle)
+        })
+    end
+end
+
+local function pushVehicle()
+    if cache.vehicle then return end
+
+    local ped = cache.ped
+    local vehicle, vehicleCoords = lib.getClosestVehicle(GetEntityCoords(ped), 3.0)
+    if not vehicle then return end
+
+    if IsEntityAttachedToEntity(cache.ped, vehicle) or not isVehicleValid(vehicle) then return end
+
+    vehicleValidityThread(vehicle)
+
+    if not pressed.e or not pressed.shift then return end
+
+    local pedCoords = GetEntityCoords(ped)
+    local boneIndex = GetPedBoneIndex(ped, 6286)
+    local dimensions = GetModelDimensions(GetEntityModel(vehicle))
+    local vehicleForwardVector = GetEntityForwardVector(vehicle)
+    local isInFront = #(vehicleCoords - pedCoords + vehicleForwardVector) < #(vehicleCoords - pedCoords - vehicleForwardVector)
+
+    if isInFront then
+        AttachEntityToEntity(ped, vehicle, boneIndex, 0.0, dimensions.y * -1 + 0.1, dimensions.z + 1.0, 0.0, 0.0, 180.0,
+            false, false, false, true, 0, true)
+    else
+        AttachEntityToEntity(ped, vehicle, boneIndex, 0.0, dimensions.y - 0.3, dimensions.z + 1.0, 0.0, 0.0, 0.0, false,
+            false, false, true, 0, true)
+    end
+
+    lib.requestAnimDict(dict)
+    TaskPlayAnim(ped, dict, 'pushcar_offcliff_m', 2.0, -8.0, -1, 35, 0, false, false, false)
```

Comment:
You can use lib.playAnim instead (just also noticed this is not on the ox docs but it is in the code)